### DISCUSSION TITLE
Upgrading to v6.x

### DISCUSSION THREAD

**Discussion by blondie63** [Laravel-Backpack/community-forum#616] [Q&A (Help)] [answered]
I tried to upgrade following official guide, at final of cmd: php artisan backpack:install i've this error:
Checking Backpack Basset installation.  

  Initializing basset check ................................................................................................................ DONE  
  Checking cache storage ................................................................................................................... DONE  
  Fetching a basset ....................................................................................................................... ERROR  

   ERROR  Error fetching the file...

following https://github.com/laravel-backpack/basset#faq i check my config that working fine with previous v5.x
the guide talk about config/backpack/basset.php but i don't have this file
I also take a look on backpack demo prj and this file does'nt exist on my local

Anyway i've tried cmd: php artisan storage:link and have this:
  ERROR  The [public/storage] link already exists.
  
 What can i do ?
 Thanks for help

**Comment by pxpm** [comment]
@blondie63 you can try publishing the config of basset: `php artisan vendor:publish --provider="Backpack\Basset\BassetServiceProvider"`

And then change the disk to `basset`. 

Let me know how it goes.

**Reply by blondie63** [reply]
do you mean change line 'default' => env('FILESYSTEM_DISK', 'local'), in filesystem.php ?

**Reply by pxpm** [reply]
No, in basset configuration as it says in the docs link you provided: 
![image](https://github.com/Laravel-Backpack/community-forum/assets/7188159/eefaf7db-2b67-47fc-96e6-b75330848631)

**Reply by blondie63** [reply]
ok, in config/backpack/basset.php file i've changed line n.8 from 'disk' => env('BASSET_DISK', 'public'),
to 'disk' => env('BASSET_DISK', 'basset'),

but the same cmd php artisan basset:check fail
what's wrong ?

**Reply by pxpm** [reply]
Maybe the `APP_URL` ? 

What url do you use to access your application in browser, and what url is set in your APP_URL in `.env` file ? 

Can you share the errors from missing files in your Developer Console in browser ? OR is everything working and this is a false positive ? 

Cheers

**Reply by blondie63** [reply]
I'm on my dev Mac
.env have:
APP_DEBUG=true
APP_URL=
SERVER_PORT=8010
my app with bkp 5.x start fine at http://localhost:8010

I created a new branch and follow steps to upg to backpack v6.x
I've published basset with cmd: php artisan vendor:publish --provider="Backpack\Basset\BassetServiceProvider"

Then i modified it as you suggested:
<?php

return [
    // development mode, assets will not be internalized
    'dev_mode' => env('BASSET_DEV_MODE', env('APP_ENV') === 'local'),

    // disk and path where to store bassets
    'disk' => env('BASSET_DISK', 'basset'),
    'path' => 'basset',

    // use cache map file (.basset)
    'cache_map' => env('BASSET_CACHE_MAP', true),

    // view paths that may use @basset
    // used to internalize assets in advance with artisan basset:internalize
    'view_paths' => [
        resource_path('views'),
    ],

    // content security policy nonce
    'nonce' => null,

    // use relative path
    'relative_paths' => env('BASSET_RELATIVE_PATHS', true),
];

but still now cmd php artisan basset:check fail
php artisan serve start so:
INFO  Server running on [http://127.0.0.1:8010].  

  Press Ctrl+C to stop the server

  2023-08-04 17:41:45 ...................................................................................................................... ~ 0s
  2023-08-04 17:41:45 /favicon.ico


But running http://localhost:8010 i've:
ERROR
500
It's not you, it's me.

Looking in Developer Console there are no errors

Did I make some mistake?

**Reply by pxpm** [reply]
Can you put `APP_URL=http://localhost:8010` ?

It should work just fine.

**Reply by blondie63** [reply]
Just tried with same result!
running php artisan storage:link

   ERROR  The [public/storage] link already exists.

because i just have one my public/storage folder.. maybe this is the problem and i need to change this name ?

I confirm this, renaming my storage folder i can run cmd: php artisan storage:link without problem
About cmd: php artisan basset:check i can run it only in another shell when web server is running..
i'm using php artisan serve to test apps..

Anyway my app still start with err 500 and i don't know if the problem is renamed storage folder..
really i don't thinks so because login page not need files from public folder..

Maybe the problem is other :(

**Reply by pxpm** [reply]
Yes, basset: check only works when the webserver is running. It attempts to get a file, and serve it as if you were using the directives on the page. 

I also use php artisan serve to test some apps, as long as I have the proper url in env, with the port included, I can access my server in http://localhost:8000 that's the same value I have in my app_url.

**Comment by blondie63** [comment]
It seems to me that this point is not very clear:
```
2. Improperly configured disk.
By default, Basset uses the Laravel public disk. For new Laravel projects, the configuration is usually correct. If you're upgrading a project and/or changed the public disk configuration, it's advised that you change the basset disk in config/backpack/basset.php to basset. The basset disk is a copy of the original Laravel public with working configurations.
```

I have these doubts:
1. my app working fine and have a /public folder with my files,  do i need to do something with this folder?
2. "The basset disk is a copy of the original Laravel public" but who make this copy ?
<img width="175" alt="Screenshot 2023-08-04 alle 18 17 27" src="https://github.com/Laravel-Backpack/community-forum/assets/1261851/23c5305c-5b1d-465f-8cc3-53dbf32c9a4d">

4. in basset.php i've this line: 'path' => 'basset', | but this is a relative path from where ? which is the absolute path ?

Thanks for help

**Comment by pxpm** [comment]
We register the `basset` disk in the provider. It's a "copy" of the public disk provided by Laravel. 

https://github.com/Laravel-Backpack/basset/blob/4858f63d4036d3054a4d578aabbff77516aad334/src/BassetServiceProvider.php#L172

**Comment by blondie63** [comment]
Hi @pxpm i've enable all possible debug settings, my app start with err 500 without any error in laravel.log
I only see this on browser:
<img width="2041" alt="cmobi-err500" src="https://github.com/Laravel-Backpack/community-forum/assets/1261851/6bfb0ca4-3c99-429f-b206-7b201b634e8f">

What you think ?

**Comment by blondie63** [comment]
HI @pxpm Would you take a second to help me with this 500 error?

**Reply by pxpm** [reply]
Sure, let me try. 

I am assuming you have `ENV=local` and `DEBUG=true`.

Do you have the error views in `resources/views/errors` ?

**Reply by blondie63** [reply]
yes i've ENV=local and DEBUG=true
<img width="195" alt="resource views error" src="https://github.com/Laravel-Backpack/community-forum/assets/1261851/8df3a1b6-7b47-4ea1-8850-52d15c748288">

**Reply by pxpm** [reply]
Can you delete them ? Or rename the errors folder to something else to keep them as backup. 

It's part of the upgrade process to remove those views. 

Let me know.

**Reply by blondie63** [reply]
Sorry for delay..
I renamed resources/views/errors to old but i've same 500 err :(

**Reply by blondie63** [reply]
@pxpm @tabacitu To try solve this 500 error i need to revert path for my public/storage folder
Basset can be configured to use an alternative custom path ?

**Comment by blondie63** [comment, marked as answer]
Hi @pxpm , finally, adding a dd($e->getMessage()); to my hadler.php i had the error info:
"The view: [backpack.ui::auth.login] was not found in any of the following view paths: [backpack.theme-coreuiv2::auth.login ], [ backpack.theme-coreuiv2::auth.lo ▶" // app/Exceptions/Handler.php:20

I think the problem was caused by the fact that during the BKP v6 installation, in step 3.2, I chose not to install any new templates, I sincerely thought I could do it later!

I then followed steps 8 and 9 and chose to use theme-coreuiv2
I moved all my customizations to ui.php
but now I no longer see my menus but the backpack default ones

What's wrong ?
